qiskit_machine_learning/datasets/gaussian.py

```diff
@@ -22,138 +22,102 @@
-                elif randomized_vector3[feat] == 2:
-                    sample_c[t_r][feat] = \
-                        algorithm_globals.random.normal(2 * 5 * np.pi / 6, sigma, None)
-
-        sample_train = [sample_a, sample_b, sample_c]
-        for lindex in range(training_size + test_size):
-            label_train[lindex] = 0
-        for lindex in range(training_size + test_size):
-            label_train[training_size + test_size + lindex] = 1
-        for lindex in range(training_size + test_size):
-            label_train[training_size + test_size + training_size + test_size + lindex] = 2
-        label_train = label_train.astype(int)
-        sample_train = np.reshape(sample_train, (3 * (training_size + test_size), n))
-        training_input = {key: (sample_train[label_train == k, :])[:training_size]
-                          for k, key in enumerate(class_labels)}
-        test_input = {key: (sample_train[label_train == k, :])[training_size:(
-                training_size + test_size)] for k, key in enumerate(class_labels)}
-
-        training_feature_array, training_label_array = features_and_labels_transform(
-            training_input, class_labels, one_hot)
-        test_feature_array, test_label_array = features_and_labels_transform(
-            test_input, class_labels, one_hot)
-
-        if plot_data:
-            try:
-                import matplotlib.pyplot as plt
-            except ImportError as ex:
-                raise MissingOptionalLibraryError(
-                    libname='Matplotlib',
-                    name='gaussian',
-                    pip_install='pip install matplotlib') from ex
-
-            for k in range(0, 3):
-                plt.scatter(sample_train[label_train == k, 0][:training_size],
-                            sample_train[label_train == k, 1][:training_size])
-
-            plt.title("Gaussians")
-            plt.show()
-
-        return training_feature_array, training_label_array, test_feature_array, test_label_array
-    else:
+    if n not in (2, 3):
         raise ValueError("Gaussian presently only supports 2 or 3 qubits")
+
+    class_labels = _generate_class_labels(n)
+    label_train, randomized_vectors, samples_train = _init_data_structures(n, test_size,
+                                                                           training_size)
+    sigma = 1
+
+    for t_r in range(training_size + test_size):
+        for feat in range(n):
+            for sample, randomized_vector in zip(samples_train, randomized_vectors):
+                sample[t_r][feat] = _calc_random_normal(randomized_vector, feat,
+                                                        sigma) if n == 2 \
+                    else _calc_random_normal_2(randomized_vector, feat, sigma)
+                # TODO unify both functions in the future when other n supported
+
+    _update_label_train(n, label_train, test_size, training_size)
```

Comment:
While I see the intent I just wonder if it goes too far. When we have method that is only called from one place and its a few lines long, for me this makes things harder to see as I have to go back and forth to methods to see what was laid out in front of me clearly before. But I do agree that there was room for improvement

Reply:
I often refer to the book by Robert C. Martin "Clean code". The essence of what he suggests there is summarized in this article (https://medium.com/swlh/clean-code-writing-functions-or-methods-4e6e53ff4ac2):

"The first rule of writing functions is functions should be small. The second rule for writing functions is that they should be even smaller. Functions that have 200 or 300 lines are kind of messy functions. They should not be longer than 20 lines and mostly less than 10 lines. Functions should be making as few arguments as possible, ideally none. Small functions are easy to understand and their intention is clear." 

But of course the line between what should be a separate function and what should not will always be blurry.